src/ape/contracts/base.py

```diff
@@ -677,6 +677,74 @@ def _mutable_methods_(self) -> Dict[str, ContractTransactionHandler]:
             # NOTE: Must raise AttributeError for __attr__ method or will seg fault
             raise AttributeError(str(err)) from err
 
+    def call_view_method(self, method_name: str, *args, **kwargs) -> Any:
+        """
+        Call a contract's view function directly using the method_name.
+        This is helpful in the scenario where the contract has a
+        method name matching an attribute of the BaseAddress class,
+        such as 'nonce' or 'balance'
+
+        Args:
+            method_name (str): The contract method name to be called
+            *args: Contract method arguments.
+            **kwargs: Transaction values, such as ``value`` or ``sender``
+
+        Returns:
+            Any: Output of smart contract view call.
+
+        """
+
+        if method_name in self._view_methods_:
+            view_handler = self._view_methods_[method_name]
+            output = view_handler(*args, **kwargs)
+            return output
+
+        elif method_name in self._mutable_methods_:
+            mutable_handler = self._mutable_methods_[method_name]
+            output = mutable_handler(*args, **kwargs)
+            return output
+
+        else:
+            # Didn't find anything that matches
+            # NOTE: `__getattr__` *must* raise `AttributeError`
+            name = self.contract_type.name or self.__class__.__name__
+            raise AttributeError(f"'{name}' has no attribute '{method_name}'.")
+
+    def invoke_transaction(self, method_name: str, *args, **kwargs) -> ReceiptAPI:
+        """
+        Call a contract's function directly using the method_name.
+        This function is for non-view function's which may change
+        contract state and will execute a transaction.
+        This is helpful in the scenario where the contract has a
+        method name matching an attribute of the BaseAddress class,
+        such as 'nonce' or 'balance'
+
+        Args:
+            method_name (str): The contract method name to be called
+            *args: Contract method arguments.
+            **kwargs: Transaction values, such as ``value`` or ``sender``
+
+        Returns:
+            ReceiptAPI: Output of smart contract interaction.
+
+        """
+
+        if method_name in self._view_methods_:
+            view_handler = self._view_methods_[method_name]
```

Comment:
I think we need to handle these as transactions since they are in the `invoke_transaction()` method.
To do this, it is something like:

```python
handler = self._view_methods_[method_name].transact
```

To ensure we are correctly doing this, we should add more tests for the following scenarios:

1. Test that shows we can call a mutable method by name in `call_view_method()` and it is treated like a call (a simulation)
2. Test that shows we can call a view method in `invoke_transaction()` and it will transact and return us a receipt, even though it is only a view method.

Reply:
Great catch, I was actually unaware of the difference between passing back a `ContractTransactionHandler` object with and without the `.transact` attached.  Will implement this change